src/Compressors/count_sketch.jl

```
@@ -0,0 +1,176 @@
+``n \\times s``, where ``s`` is the user-specified compression dimension. Each row of ``S`` is 
+generated independently using the following steps:
+
+1. Randomly select an integer between 1 and ``s`` to determine the column position of the nonzero entry.
+2. Assign this entry a value of either +1 or -1, chosen uniformly at random.
+3. Set all other entries in the row to zero.
+
+In this case, each row of S has exactly one nonzero entry. 
+The compressed matrix is then formed by multiplying S A (for left compression) or A S (for right compression).
+
+# Fields
+- `cardinality::Cardinality`, the direction the compression matrix is intended to be
+    applied to a target matrix or operator. Values allowed are `Left()` or `Right()`.
+- `compression_dim::Int64`, the target compression dimension. Referred to as ``s`` in the
+    mathematical description.
+- `type::Type{<:Number}`, the type of the elements in the compressor.
+
+# Constructor
+
+    CountSketch(;carinality=Left(), compression_dim=2, type=Float64)
+
+## Keywords
+- `carinality::Cardinality`, the direction the compression matrix is intended to be
+    applied to a target matrix or operator. Values allowed are `Left()` or `Right()`.
+    By default `Left()` is chosen.
+- `compression_dim`, the target compression dimension. Referred to as ``s`` in the
+    mathemtical description. By default this is set to 2.
+- `type::Type{<:Number}`, the type of the elements in the compressor. By default is set
+    to Float64.
+
+## Returns
+- A `CountSketch` object.
+
+## Throws
+- `ArgumentError` if `compression_dim` is non-positive
+"""
+struct CountSketch <: Compressor
+    cardinality::Cardinality
+    compression_dim::Int64
+    type::Type{<:Number}
+    # Check on the compression dimension
+    function CountSketch(cardinality, compression_dim, type)
+        if compression_dim <= 0
+            throw(ArgumentError("Field 'compression_dim' must be positive."))
+        end
```

Should throw an error if `cardinality==Undef()`.